src/main/java/ch/njol/skript/sections/EffSecFireworkLaunch.java

```diff
@@ -0,0 +1,175 @@
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Trigger;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.util.Getter;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+
+@Name("Launch Firework")
+@Description({
+	"Launch firework effects at the given location(s).",
+	"This can be used as an effect and as a section.",
+	"If it is used as a section, the section is run before the entity is added to the world.",
+	"You can modify the firework in this section, using for example <code>event-entity</code>.",
+	"Do note that other event values, such as <code>player</code>, won't work in this section."
+})
+@Examples({
+	"launch ball large coloured red, purple and white fading to light green and black at player's location with duration 1",
+	"",
+	"command /firework:",
+	"\ttrigger:",
+		"\t\tlaunch a firework with effects ball large coloured red at player:",
+			"\t\t\tset metadata value \"cancel damage\" of event-entity to true",
+	"",
+	"# Firework launch section example",
+	"on damage:",
+		"\tdamage cause is entity explosion",
+		"\tmetadata value \"cancel damage\" of event-projectile is true",
+		"\tcancel event",
+})
+@Since("2.4, INSERT VERSION (section)")
+public class EffSecFireworkLaunch extends EffectSection {
+
+	public static class FireworkSectionLaunchEvent extends Event {
+
+		private final Firework firework;
+
+		public FireworkSectionLaunchEvent(Firework firework) {
+			this.firework = firework;
+		}
+
+		public Firework getFirework() {
+			return firework;
+		}
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+
+	}
+
+	static {
+		Skript.registerSection(EffSecFireworkLaunch.class, "(launch|deploy) [[a] firework [with effect[s]]] %fireworkeffects% at %locations% [([with] (duration|power)|timed) %-number%]");
+		EventValues.registerEventValue(FireworkSectionLaunchEvent.class, Firework.class, new Getter<Firework, FireworkSectionLaunchEvent>() {
+			@Override
+			public Firework get(FireworkSectionLaunchEvent fireworkLaunchEvent) {
+				return fireworkLaunchEvent.getFirework();
+			}
+		}, EventValues.TIME_NOW);
+	}
+
+	private Expression<FireworkEffect> effects;
+	private Expression<Location> locations;
+
+	@Nullable
+	public static Entity lastSpawned;
+
+	@Nullable
+	private Expression<Number> power;
+
+	@Nullable
+	private Trigger trigger;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult,
+			@Nullable SectionNode sectionNode, @Nullable List<TriggerItem> triggerItems) {
+
+		effects = (Expression<FireworkEffect>) exprs[0];
+		locations = (Expression<Location>) exprs[1];
+		power = (Expression<Number>) exprs[2];
+
+		if (sectionNode != null)
+			trigger = loadCode(sectionNode, "firework launch", FireworkSectionLaunchEvent.class);
```

Comment:
It would be good to include the logic from EffSecSpawn that forbids delays.
https://github.com/SkriptLang/Skript/blob/0355df1a07c027ca2fb5355eeaa57378d524f813/src/main/java/ch/njol/skript/sections/EffSecSpawn.java#L125-L133